kernel/src/transaction.rs

```diff
@@ -215,6 +215,18 @@ impl Transaction {
         Expression::struct_from(fields)
     }
 
+    fn generate_logical_to_physical_with_schema(&self, schema: &SchemaRef) -> Expression {
```

Comment:
Let's just add a `schema` arg to `generate_logical_to_physical`, and pass `snapshot_schema` in `get_write_context`